test/tools/paths.d

```diff
@@ -0,0 +1,76 @@
+module tools.paths;
+
+import std.file : exists;
+import std.path : buildNormalizedPath, buildPath, dirName, setExtension;
+import std.process : environment;
+
+version (Posix)
+    enum exeExtension = "";
+else version (Windows)
+    enum exeExtension = ".exe";
+
+version(Windows)
+    enum os ="windows";
+else version(OSX)
+    enum os ="osx";
+else version(linux)
+    enum os ="linux";
+else version(FreeBSD)
+    enum os ="freebsd";
+else version(OpenBSD)
+    enum os ="openbsd";
+else version(NetBSD)
+    enum os ="netbsd";
+else version(DragonFlyBSD)
+    enum os ="dragonflybsd";
+else version(Solaris)
+    enum os ="solaris";
+else version(SunOS)
+    enum os ="solaris";
+else
+    static assert(0, "Unrecognized or unsupported OS.");
+
+enum projectRootDir = __FILE_FULL_PATH__.dirName.buildNormalizedPath("..", "..");
+enum generatedDir = projectRootDir.buildPath("generated");
+enum resultsDir = testPath("test_results");
+
+enum dmdFilename = "dmd".setExtension(exeExtension);
+
+alias testPath = path => projectRootDir.buildPath("test", path);
+
+string build()
```

Comment:
`buildType` or maybe simply `BUILD` (like in Make)?

Reply:
It was called `build` before: https://github.com/dlang/dmd/blob/a60e9e25f075af0179f521e8e6864d892996a713/test/run.d#L254